lib/parse.js

```diff
@@ -919,13 +925,21 @@ function parse($TEXT, options) {
         if (!is("punc", ";")) {
             init = is("keyword", "var")
                 ? (next(), var_(true))
-                : expression(true, true);
+                : is("keyword", "let")
```

Comment:
If you see a `let` here, you have to look ahead for either a `[`, `{`, or identifier to know it's a declaration. `let.a` is valid in this position.